google/cloud/aiplatform/datasets.py

```diff
@@ -174,19 +178,40 @@ def create(
         """
 
         # Validate that source and import schema are passed together or not at all
-        if bool(source) ^ bool(import_schema_uri):
+        if bool(gcs_source) ^ bool(import_schema_uri):
             raise ValueError(
-                "Please provide both source and import_schema_uri to import data or omit both."
+                "Please provide both GCS source and import_schema_uri to import data or omit both."
             )
 
         api_client = cls._instantiate_client(location=location, credentials=credentials)
 
+        # If this is tabular enrich the dataset metadata with source
+        # TODO: Use interfaces to abstract away gcs and bq specific logic
+        dataset_metadata = {}
+        is_tabular_dataset_metadata = (
+            metadata_schema_uri == schema.dataset.metadata.tabular
+        )
+        if is_tabular_dataset_metadata:
+            if gcs_source:
+                dataset_metadata = {"input_config": {"gcs_source": {"uri": gcs_source}}}
+            elif bq_source:
+                dataset_metadata = {
+                    "input_config": {"bigquery_source": {"uri": bq_source}}
+                }
+
+        # TODO: Remove this and let the error propagate from up from downstream?
```

Comment:
I'm about to remove this until more discussion regarding the validation. See PR description. Thoughts?